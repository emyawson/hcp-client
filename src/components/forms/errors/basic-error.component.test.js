import React from 'react';
import { shallow } from 'enzyme';

import { BasicError, showFieldError } from './basic-error.component';

const mockFieldTouchedProps = {
  touched: true,
  focus: false,
};
const mockFieldActiveProps = {
  touched: true,
  focus: true,
};

describe('Basic Form Error', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<BasicError />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show errors for dirty fields', () => {
    expect(showFieldError(mockFieldTouchedProps)).toBeTruthy();
  });
  it('should suppress errors for focused fields', () => {
    expect(showFieldError(mockFieldActiveProps)).toBeFalsy();
  });
});
