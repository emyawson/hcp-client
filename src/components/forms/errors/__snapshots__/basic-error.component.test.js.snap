// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Form Error should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BasicError />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "color": "trafficRed",
      "fontSize": "0.7rem",
      "messages": Object {
        "isRequired": "required",
      },
      "model": undefined,
      "show": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "color": "trafficRed",
        "fontSize": "0.7rem",
        "messages": Object {
          "isRequired": "required",
        },
        "model": undefined,
        "show": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
